java/ql/test/query-tests/security/CWE-489/DebuggableAttributeEnabledTest.ql

```diff
@@ -0,0 +1,22 @@
+import java
+import semmle.code.xml.AndroidManifest
+import TestUtilities.InlineExpectationsTest
+
+class DebuggableAttributeTrueTest extends InlineExpectationsTest {
+  DebuggableAttributeTrueTest() { this = "DebuggableAttributeEnabledTest" }
+
+  override string getARelevantTag() { result = "hasDebuggableAttributeEnabled" }
+
+  override predicate hasActualResult(Location location, string element, string tag, string value) {
+    tag = "hasDebuggableAttributeEnabled" and
+    exists(AndroidXmlAttribute androidXmlAttr |
+      androidXmlAttr.getName() = "debuggable" and
+      androidXmlAttr.getValue() = "true" and
+      not androidXmlAttr.getLocation().getFile().getRelativePath().matches("%build%")
```

Comment:
To avoid repetition, can you think of a way of turning this logic into a predicate of `AndroidApplicationXmlElement` in `AndroidManifest.qll`, and then use that in both the query file and this test?

Reply:
Also, do you think it makes sense to add a test case for the `build` directory exclusion?

Reply:
Updated in commit [171a47e](https://github.com/github/codeql/pull/9939/commits/171a47e816feaeda3533f793510eb12bdc76bf0f).